subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c

```diff
@@ -25,30 +25,44 @@ LOG_MODULE_DECLARE(mcumgr_img_grp, CONFIG_MCUMGR_GRP_IMG_LOG_LEVEL);
 #define SLOT1_PARTITION		slot1_partition
 #define SLOT2_PARTITION		slot2_partition
 #define SLOT3_PARTITION		slot3_partition
+#define SLOT4_PARTITION		slot4_partition
+#define SLOT5_PARTITION		slot5_partition
 
 BUILD_ASSERT(CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 1 ||
-	     (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 2 &&
+	     (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER >= 2 &&
 	      FIXED_PARTITION_EXISTS(SLOT2_PARTITION) &&
-	      FIXED_PARTITION_EXISTS(SLOT3_PARTITION)),
+	      FIXED_PARTITION_EXISTS(SLOT3_PARTITION)) ||
+	     (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 3 &&
+	      FIXED_PARTITION_EXISTS(SLOT4_PARTITION) &&
+	      FIXED_PARTITION_EXISTS(SLOT5_PARTITION)
+	      ),
 	     "Missing partitions?");
 
 #if defined(CONFIG_MCUMGR_GRP_IMG_DIRECT_UPLOAD) &&		\
 	!(CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER > 1)
-/* In case when direct upload is enabled, slot2 and slot3 are optional
- * as long as there is support for one application image only.
+/* In case when direct upload is enabled, slot2, slot3, slot4 and slot5
+ *  are optional as long as there is support for one application image only.
  */
 #define ADD_SLOT_2_CONDITION FIXED_PARTITION_EXISTS(SLOT2_PARTITION)
 #define ADD_SLOT_3_CONDITION FIXED_PARTITION_EXISTS(SLOT3_PARTITION)
+#define ADD_SLOT_4_CONDITION FIXED_PARTITION_EXISTS(SLOT4_PARTITION)
+#define ADD_SLOT_5_CONDITION FIXED_PARTITION_EXISTS(SLOT5_PARTITION)
 #elif (CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER > 1)
 /* For more than one application image slot2 and slot3 are required. */
 #define ADD_SLOT_2_CONDITION 1
 #define ADD_SLOT_3_CONDITION 1
+#if CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER == 3
+#define ADD_SLOT_4_CONDITION 1
```

Comment:
A long term future goal should be to remove all of this and have "metadata" specified somehow, a main application might have 2 slots in the primary image, or actually might only have 1 slot, but might then have mismatched slots in other images, maybe 1 slot for an image for another device or core, maybe 2 slots for something else, but we need some way to have the freedom and flexibility to have odd or equal number of slots per image.
And to have it generated so we don't need to update this file each time a new slot/image is added